components/eamxx/src/physics/mam4/aerosol_microphysics.hpp

```diff
@@ -0,0 +1,247 @@
+                       const view_2d&       pdel,
+                       const view_1d_const& pblh,
+                       const view_2d&       q_soag,
+                       const view_2d&       q_h2so4,
+                       const view_2d&       q_nh3,
+                       const view_2d&       q_aitken_so4) {
+      ncol_ = ncol;
+      nlev_ = nlev;
+      T_mid_ = T_mid;
+      p_mid_ = p_mid;
+      qv_ = qv;
+      height_ = height;
+      pdel_ = pdel;
+      pblh_ = pblh;
+      q_soag_ = q_soag;
+      q_h2so4_ = q_h2so4;
+      q_nh3_ = q_nh3;
+      q_aitken_so4_ = q_aitken_so4;
+    } // set_variables
+  }; // MAM4AerosolMicrophysics::Preprocess
+
+  // Postprocessing functor
+  struct Postprocess {
+    Postprocess() = default;
+
+    KOKKOS_INLINE_FUNCTION
+    void operator()(const Kokkos::TeamPolicy<KT::ExeSpace>::member_type& team) const {
+      // Copy the updated aerosol tracers back into the tracer array.
+      // FIXME
+
+      // After these updates, all tracers are converted from dry mmr to wet mmr
+      const int i = team.league_rank();
+      Kokkos::parallel_for(Kokkos::TeamThreadRange(team, nlev_), [&](const int k) {
+        // Here we convert our dry mmrs back to wet mmrs for EAMxx.
+        // NOTE: calculate_wetmmr_from_drymmr takes 2 arguments:
+        // 1. dry mmr
+        // 2. "dry" water vapor mixing ratio
+        q_soag_(i,k)  = PF::calculate_wetmmr_from_drymmr(q_soag_(i,k), qv_(i,k));
+        q_h2so4_(i,k) = PF::calculate_wetmmr_from_drymmr(q_h2so4_(i,k), qv_(i,k));
+        q_nh3_(i,k)   = PF::calculate_wetmmr_from_drymmr(q_nh3_(i,k), qv_(i,k));
+
+        q_aitken_so4_(i,k) = PF::calculate_wetmmr_from_drymmr(q_aitken_so4_(i,k), qv_(i,k));
+
+        qv_(i,k) = PF::calculate_wetmmr_from_drymmr(qv_(i,k), qv_(i,k));
+      });
+      team.team_barrier();
+    } // operator()
+
+    // Local variables
+    int ncol_, nlev_;
+    view_2d qv_;
+
+    // local aerosol-related gases
+    view_2d       q_soag_;  // secondary organic aerosol gas [kg gas/kg dry air]
+    view_2d       q_h2so4_; // H2SO3 gas [kg/kg dry air]
+    view_2d       q_nh3_;   // NH3 gas [kg/kg dry air]
+
+    // local aerosols (more to appear as we improve this atm process)
+    view_2d       q_aitken_so4_; // SO4 aerosol in aitken mode [kg/kg dry air]
+
+    // assigns local variables
+    void set_variables(const int ncol,
+                       const int nlev,
+                       const view_2d& qv,
+                       const view_2d& q_soag,
+                       const view_2d& q_h2so4,
+                       const view_2d& q_nh3,
+                       const view_2d& q_aitken_so4) {
+      ncol_ = ncol;
+      nlev_ = nlev;
+      qv_ = qv;
+      q_soag_ = q_soag;
+      q_h2so4_ = q_h2so4;
+      q_nh3_ = q_nh3;
+      q_aitken_so4_ = q_aitken_so4;
+    } // set_variables
+  }; // MAM4AerosolMicrophysics::Postprocess
+
+  // MAM4 aerosol particle size description
+  mam4::AeroConfig aero_config_;
```

Comment:
Is there a standard meaning to a variable ending in an underscore? Is this standard across all of EAMxx, or just within MAMxx? This is probably a matter of me not spending enough time in code rather than a real problem.

Reply:
Ah, it's just something I use to indicate that a variable or method is `private` or `protected`. I don't think we have C++ style rules for EAMxx yet, or if we do, I'm not aware of them (or have forgotten about them!). It's easy enough for us to change these to fit a spec. But I like some indication that something belongs to an object, because it makes analyzing code in methods easier.

Reply:
I usually use `m_` for member variables, and `s_` for static member variables. I've seen the trailing underscore used in the opposite way in other libs, to distinguish the input var `foo_` from the stored var `foo`, usually in constructor/setter methods, so that the shorter version is available in other methods. I personally dislike no prefix/suffix for member vars, since you never know inside an impl whether the var is local or not (but i'm ok with it in case of tiny small POD structs).

I think we briefly discussed coding conventions at the beginning, but the consensus was, IIRC, that we didn't want ppl to get lost amidst conventions.

Reply:
Happy to go with the flow here, too. It's pretty easy to make these kinds of changes whenever we need to, since we're already following a convention.

Reply:
No need to, since you did implement some "class-member-marker" (the underscore). We can do a formatting PR if/when we ever decide is important/needed.